mobile/mobile-core/devices-service.ts

```diff
@@ -450,8 +535,14 @@ export class DevicesService implements Mobile.IDevicesService {
 		await device.applicationManager.tryStartApplication(packageName);
 	}
 
-	private hasDevice(identifier: string): boolean {
-		return _.some(this.getDeviceInstances(), (device: Mobile.IDevice) => { return device.deviceInfo.identifier === identifier; });
+	/**
+	 * Returns true if there's a running device with specified identifier.
+	 * @param identifier parameter passed by the user to --device flag
+	 */
+	private async hasDevice(identifier: string): Promise<boolean> {
```

Comment:
As per the comment of the method I would make it more self-descriptive as "hasRunningDevice"